exp/lighthorizon/services/main.go

```diff
@@ -0,0 +1,192 @@
+}
+
+type searchCallback func(archive.LedgerTransaction, *xdr.LedgerHeader) (finished bool, err error)
+
+func (os *OperationsService) GetOperationsByAccount(ctx context.Context, cursor int64, limit int64, accountId string) ([]common.Operation, error) {
+	ops := []common.Operation{}
+	opsCallback := func(tx archive.LedgerTransaction, ledgerHeader *xdr.LedgerHeader) (bool, error) {
+		for operationOrder, op := range tx.Envelope.Operations() {
+			opParticipants, opParticipantErr := os.Config.Archive.GetOperationParticipants(tx, op, operationOrder+1)
+			if opParticipantErr != nil {
+				return false, opParticipantErr
+			}
+			if _, foundInOp := opParticipants[accountId]; foundInOp {
+				ops = append(ops, common.Operation{
+					TransactionEnvelope: &tx.Envelope,
+					TransactionResult:   &tx.Result.Result,
+					LedgerHeader:        ledgerHeader,
+					TxIndex:             int32(tx.Index),
+					OpIndex:             int32(operationOrder + 1),
+				})
+				if int64(len(ops)) == limit {
+					return true, nil
+				}
+			}
+		}
+		return false, nil
+	}
+
+	if err := searchTxByAccount(ctx, cursor, accountId, os.Config, opsCallback); err != nil {
+		return nil, err
+	}
+
+	return ops, nil
+}
+
+func (ts *TransactionsService) GetTransactionsByAccount(ctx context.Context, cursor int64, limit int64, accountId string) ([]common.Transaction, error) {
+	txs := []common.Transaction{}
+
+	txsCallback := func(tx archive.LedgerTransaction, ledgerHeader *xdr.LedgerHeader) (bool, error) {
+		txs = append(txs, common.Transaction{
+			TransactionEnvelope: &tx.Envelope,
+			TransactionResult:   &tx.Result.Result,
+			LedgerHeader:        ledgerHeader,
+			TxIndex:             int32(tx.Index),
+		})
+		if int64(len(txs)) == limit {
+			return true, nil
+		}
+		return false, nil
+	}
+
+	if err := searchTxByAccount(ctx, cursor, accountId, ts.Config, txsCallback); err != nil {
+		return nil, err
+	}
+	return txs, nil
+}
+
+func searchTxByAccount(ctx context.Context, cursor int64, accountId string, config Config, callback searchCallback) error {
+	// Skip the cursor ahead to the next active checkpoint for this account
+	nextCheckpoint, err := getAccountNextCheckpointCursor(accountId, cursor, config.IndexStore)
+	if err != nil {
+		if err == io.EOF {
+			return nil
+		}
+		return err
+	}
+	log.Debugf("Searching index by account %v starting at checkpoint cursor %v", accountId, nextCheckpoint)
+
+	for {
+		startingCheckPointLedger := cursorLedger(nextCheckpoint)
+		ledgerSequence := startingCheckPointLedger
+
+		for (ledgerSequence - startingCheckPointLedger) < 64 {
+			ledger, ledgerErr := config.Archive.GetLedger(ctx, ledgerSequence)
+			if ledgerErr != nil {
+				return errors.Wrapf(ledgerErr, "ledger export state is out of sync, missing ledger %v from checkpoint %v", ledgerSequence, ledgerSequence/64)
+			}
+
+			reader, readerErr := config.Archive.NewLedgerTransactionReaderFromLedgerCloseMeta(config.Passphrase, ledger)
+			if readerErr != nil {
+				return readerErr
```

Comment:
nit: error verbosity?
```suggestion
				return errors.Wrapf(readerErr, "failed reading ledger %d", ledgerSequence)
```